src/github.com/getlantern/flashlight/util/http.go

```diff
@@ -80,24 +80,37 @@ func httpClient(rootCA string, proxyAddr string, persistent bool) (*http.Client,
 			proxyAddr = host + proxyAddr
 		}
 
-		var ip net.IP
-		if ip = net.ParseIP(host); ip != nil {
-			if ip.IsLoopback() {
-				log.Debugf("Waiting for loopback proxy server to came online...")
-				// Waiting for proxy server to came online.
-				err := waitforserver.WaitForServer("tcp", proxyAddr, 60*time.Second)
-				if err != nil {
-					// Instead of finishing here we just log the error and continue, the client
-					// we are going to create will surely fail when used and return errors,
-					// those errors should be handled by the code that depends on such client.
-					log.Errorf("Proxy never came online at %v: %q", proxyAddr, err)
-				}
+		if isLoopback(host) {
+			log.Debugf("Waiting for loopback proxy server to came online...")
+			// Waiting for proxy server to came online.
+			err := waitforserver.WaitForServer("tcp", proxyAddr, 60*time.Second)
+			if err != nil {
+				// Instead of finishing here we just log the error and continue, the client
+				// we are going to create will surely fail when used and return errors,
+				// those errors should be handled by the code that depends on such client.
+				log.Errorf("Proxy never came online at %v: %q", proxyAddr, err)
 			}
+			log.Debugf("Connected to proxy on localhost")
```

Comment:
Oh, I already merged, but a quick note: it's a little weird that if the proxy never comes online, we log the error (with the promise of failing later on) followed by a debug message saying we are connected. But not really a big deal.